src/main/java/bio/terra/service/snapshot/flight/export/SnapshotExportCreateParquetFilesStep.java

```diff
@@ -37,13 +38,20 @@ public SnapshotExportCreateParquetFilesStep(
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    String flightId = context.getFlightId();
     FlightMap workingMap = context.getWorkingMap();
     Snapshot snapshot = snapshotService.retrieve(snapshotId);
     GoogleBucketResource exportBucket =
         workingMap.get(SnapshotWorkingMapKeys.SNAPSHOT_EXPORT_BUCKET, GoogleBucketResource.class);
 
-    List<String> paths =
-        bigQueryPdao.exportTableToParquet(snapshot, exportBucket, context.getFlightId());
+    final List<String> paths;
+    Boolean exportGsPathsInput =
+        context.getInputParameters().get(JobMapKeys.EXPORT_GSPATHS.getKeyName(), Boolean.class);
+    if (Boolean.TRUE.equals(exportGsPathsInput)) {
```

Comment:
This seems needlessly verbose, no?  Don't we just want
```suggestion
    if (exportGsPathsInput) {
```

Reply:
If `exportGsPath` is null, then an NPE will be thrown

Reply:
but if you make it a primitive `boolean` there is no null

Reply:
FWIW, I really dislike this pattern as well. Maybe JavaScript is onto something with its falsly `null`s.

Reply:
haha right?